open_seq2seq/encoders/tacotron2_encoder.py

```diff
@@ -0,0 +1,306 @@
+# Copyright (c) 2018 NVIDIA Corporation
+from __future__ import absolute_import, division, print_function
+from __future__ import unicode_literals
+from six.moves import range
+
+import tensorflow as tf
+from tensorflow.contrib.cudnn_rnn.python.ops import cudnn_rnn_ops
+from tensorflow.contrib.rnn import LSTMStateTuple
+from tensorflow.python.framework import ops
+
+from .encoder import Encoder
+
+def conv1d_bn_actv(name, inputs, filters, kernel_size, activation_fn, strides,
+                   padding, regularizer, training, use_bias, data_format, 
+                   enable_bn, bn_momentum, bn_epsilon):
+  """Helper function that applies 1-D convolution, batch norm and activation."""
+  conv = tf.layers.conv1d(
+    name="{}".format(name),
+    inputs=inputs,
+    filters=filters,
+    kernel_size=kernel_size,
+    strides=strides,
+    padding=padding,
+    kernel_regularizer=regularizer,
+    use_bias=use_bias,
+    data_format=data_format,
+  )
+  output = conv
+  if enable_bn:
+    bn = tf.layers.batch_normalization(
+      name="{}/bn".format(name),
+      inputs=conv,
+      gamma_regularizer=regularizer,
+      training=training,
+      axis=-1 if data_format == 'channels_last' else 1,
+      momentum=bn_momentum,
+      epsilon=bn_epsilon,
+    )
+    output = bn
+  if activation_fn is not None:
+    output = activation_fn(output)
+  return output
+
+def rnn_cell(rnn_cell_dim, layer_type, dropout_keep_prob=1.0):
```

Comment:
Can you put this function in some other utils file (similar as done with conv_bn_actv in wave2letter PR), so that it can be reused across ds2 and your encoders without copying the code?